mlfinlab/labeling/labeling.py

```diff
@@ -0,0 +1,295 @@
+"""
+Logic regarding labeling from chapter 3. In particular the Triple Barrier Method and Meta-Labeling.
+"""
+
+import numpy as np
+import pandas as pd
+
+# Snippet 3.1, page 44, Daily Volatility Estimates
+from mlfinlab.util.multiprocess import mp_pandas_obj
+
+
+def get_daily_vol(close, lookback=100):
+    """
+    Snippet 3.1, page 44, Daily Volatility Estimates
+
+    Computes the daily volatility at intraday estimation points.
+
+    In practice we want to set profit taking and stop-loss limits that are a function of the risks involved
+    in a bet. Otherwise, sometimes we will be aiming too high (tao ≫ sigma_t_i,0), and sometimes too low
+    (tao ≪ sigma_t_i,0 ), considering the prevailing volatility. Snippet 3.1 computes the daily volatility
+    at intraday estimation points, applying a span of lookback days to an exponentially weighted moving
+    standard deviation.
+
+    See the pandas documentation for details on the pandas.Series.ewm function.
+
+    Note: This function is used to compute dynamic thresholds for profit taking and stop loss limits.
+
+    :param close: Closing prices
+    :param lookback: lookback period to compute volatility
+    :return: series of daily volatility value
+    """
+    # daily vol re-indexed to close
+    df0 = close.index.searchsorted(close.index - pd.Timedelta(days=1))
+    df0 = df0[df0 > 0]
+    df0 = (pd.Series(close.index[df0 - 1], index=close.index[close.shape[0] - df0.shape[0]:]))
+
+    df0 = close.loc[df0.index] / close.loc[df0.values].values - 1  # daily returns
+    df0 = df0.ewm(span=lookback).std()
+    return df0
+
+
+# Snippet 3.2, page 45, Triple Barrier Labeling Method
+def apply_pt_sl_on_t1(close, events, pt_sl, molecule):  # pragma: no cover
+    """
+    Snippet 3.2, page 45, Triple Barrier Labeling Method
+
+    This function applies the triple-barrier labeling method. It works on a set of
+    datetime index values (molecule). This allows the program to parallelize the processing.
+
+    Mainly it returns a DataFrame of timestamps regarding the time when the first barriers were reached.
+
+    :param close: (series) close prices
+    :param events: (series) of indices that signify "events" (see get_t_events function
```

Comment:
Update docstring with new function name for cusum_filter

Reply:
Well spotted. Thank you